contrib/hadoop-store-builder/src/java/voldemort/store/readonly/fetcher/HdfsFetcher.java

```diff
@@ -215,85 +215,40 @@ private Configuration getConfiguration(String sourceFileUrl, String hadoopConfig
-            for(int attempt = 0; attempt < maxAttempts; attempt++) {
-                boolean isValidFilesystem = false;
-
-                if(!new File(HdfsFetcher.keytabPath).exists()) {
-                    logger.error("Invalid keytab file path. Please provide a valid keytab path");
-                    throw new VoldemortException("Error in getting Hadoop filesystem. Invalid keytab file path.");
-                }
-
-                /*
-                 * The Hadoop path for getting a Filesystem object in a
-                 * privileged doAs block is not thread safe. This might be
-                 * causing intermittent NPE exceptions. Adding a synchronized
-                 * block.
-                 */
-                synchronized(this) {
-                    /*
-                     * First login using the specified principal and keytab file
-                     */
-                    UserGroupInformation.setConfiguration(config);
-                    UserGroupInformation.loginUserFromKeytab(HdfsFetcher.kerberosPrincipal,
-                                                             HdfsFetcher.keytabPath);
-
-                    /*
-                     * If login is successful, get the filesystem object. NOTE:
-                     * Ideally we do not need a doAs block for this. Consider
-                     * removing it in the future once the Hadoop jars have the
-                     * corresponding patch (tracked in the Hadoop Apache
-                     * project: HDFS-3367)
-                     */
-                    try {
-                        logger.info("I've logged in and am now Doasing as "
-                                    + UserGroupInformation.getCurrentUser().getUserName());
-                        fs = UserGroupInformation.getCurrentUser()
-                                                 .doAs(new PrivilegedExceptionAction<FileSystem>() {
-
-                                                     @Override
-                                                     public FileSystem run() throws Exception {
-                                                         FileSystem fs = path.getFileSystem(config);
-                                                         return fs;
-                                                     }
-                                                 });
-                        isValidFilesystem = true;
-                    } catch(InterruptedException e) {
-                        logger.error(e.getMessage(), e);
-                    } catch(Exception e) {
-                        logger.error("Got an exception while getting the filesystem object: ");
-                        logger.error("Exception class : " + e.getClass());
-                        e.printStackTrace();
-                        for(StackTraceElement et: e.getStackTrace()) {
-                            logger.error(et.toString());
+        // TODO: Remove this for loop if we don't see the errors anymore.
```

Comment:
This code is not thread safe. Previously this was done on a synchronized this block.